drivers/pinctrl/pinctrl_nrf.c

```diff
@@ -8,6 +8,25 @@
 
 #include <hal/nrf_gpio.h>
 
+#if DT_HAS_COMPAT_STATUS_OKAY(nordic_nrf_uart)
+#define NRF_UART_TYPE NRF_UART_Type
+#elif DT_HAS_COMPAT_STATUS_OKAY(nordic_nrf_uarte)
+#define NRF_UART_TYPE NRF_UARTE_Type
+/* provide compatible macros if not available */
+#ifndef PSELTX
+#define PSELTXD PSEL.TXD
+#endif
+#ifndef PSELRXD
+#define PSELRXD PSEL.RXD
+#endif
+#ifndef PSELRTS
+#define PSELRTS PSEL.RTS
+#endif
+#ifndef PSELCTS
+#define PSELCTS PSEL.CTS
+#endif
```

Comment:
@anangl I assume this could be handled as part of nrfx

Reply:
I think so. It looks like nrfx HALs for UART and UARTE lack functions for setting TXD and RXD pins separately (same goes for RTS and CTS). With those, you would not need to access registers directly here.
For now, you could also use a macro here with dual implementation, like:
```
#define SET_PSEL(uart, line) uart->PSEL##line
```
and for UARTE:
```
#define SET_PSEL(uart, line) uart->PSEL.line
```

Reply:
Thanks, I've used your suggestion.